src/update/UpdateDataParser.cpp

```diff
@@ -0,0 +1,152 @@
+/*!********************************************************************
+ Audacity: A Digital Audio Editor
+
+ @file UpdateDataParser.cpp
+ @brief Declare a class that parse update server data format.
+
+ Anton Gerasimov
+ **********************************************************************/
+
+#include "UpdateDataParser.h"
+
+#include "xml/XMLFileReader.h"
+
+UpdateDataParser::UpdateDataParser()
+{}
+
+UpdateDataParser::~UpdateDataParser()
+{}
+
+bool UpdateDataParser::Parse(const VersionPatch::UpdateDataFormat& updateData, VersionPatch* versionPatch)
+{
+    XMLFileReader xmlReader;
+
+    mVersionPatch = versionPatch;
+    auto ok = xmlReader.ParseString(this, updateData);
+    mVersionPatch = nullptr;
```

Comment:
If I wanted to be really strict about RAII for cleanup of the side effects, making no assumptions here that ParseString won't throw, then I would write, using a class from MemoryX.h:

`ValueRestorer<VersionPatch*> setter{ mVersionPatch, versionPatch };`

Instead of the two assignments.